src/mac-now-playing/MacNowPlaying.mm

```diff
@@ -0,0 +1,267 @@
+/* file, You can obtain one at https://mozilla.org/MPL/2.0/.                 */
+/*                                                                           */
+/*****************************************************************************/
+
+#import <Cocoa/Cocoa.h>
+#import <Foundation/Foundation.h>
+#import <MediaPlayer/MediaPlayer.h>
+#include <fstream>
+#include <iostream>
+#include <istream>
+
+#include <libaudcore/drct.h>
+#include <libaudcore/hook.h>
+#include <libaudcore/i18n.h>
+#include <libaudcore/plugin.h>
+#include <libaudcore/probe.h>
+
+#define EXPORT __attribute__((visibility("default")))
+
+class MacNowPlayingPlugin : public GeneralPlugin
+{
+public:
+    static const char about[];
+    static constexpr PluginInfo info = {N_("Mac Now Playing")};
+
+    constexpr MacNowPlayingPlugin() : GeneralPlugin(info, false) {}
+
+    bool init();
+    void cleanup();
+};
+
+EXPORT MacNowPlayingPlugin aud_plugin_instance;
+
+static void update_metadata(void * data, void * user);
+
+static void update_player_time()
+{
+    if (!aud_drct_get_ready())
+    {
+        return;
+    }
+    MPNowPlayingInfoCenter * center = [MPNowPlayingInfoCenter defaultCenter];
+    double rate =
+        [center playbackState] == MPNowPlayingPlaybackStatePlaying ? 1. : 0.;
+    int current_time = aud_drct_get_time() / 1000;
+    NSDictionary<NSString *, id> * info = [center nowPlayingInfo];
+    NSMutableDictionary<NSString *, id> * minfo = [info mutableCopy];
+    minfo[MPNowPlayingInfoPropertyPlaybackRate] =
+        [[NSNumber alloc] initWithDouble:rate];
+    minfo[MPNowPlayingInfoPropertyElapsedPlaybackTime] =
+        [[NSNumber alloc] initWithInt:current_time];
+    center.nowPlayingInfo = minfo;
+}
+
+static void remote_command_setup_handler()
+{
+    // No need to change the now playing status since it will be cought be the
+    // hook.
+
+    MPRemoteCommandCenter * commandCenter =
+        [MPRemoteCommandCenter sharedCommandCenter];
+
+    // Play
+    [[commandCenter playCommand] setEnabled:YES];
+    [[commandCenter playCommand]
+        addTargetWithHandler:^MPRemoteCommandHandlerStatus(
+            MPRemoteCommandEvent * _Nonnull event) {
+          aud_drct_play();
+          return MPRemoteCommandHandlerStatusSuccess;
+        }];
+
+    // Pause
+    [[commandCenter pauseCommand] setEnabled:YES];
+    [[commandCenter pauseCommand]
+        addTargetWithHandler:^MPRemoteCommandHandlerStatus(
+            MPRemoteCommandEvent * _Nonnull event) {
+          aud_drct_pause();
+
+          return MPRemoteCommandHandlerStatusSuccess;
+        }];
+    // Toggle playpause
+    [[commandCenter togglePlayPauseCommand] setEnabled:YES];
+    [[commandCenter togglePlayPauseCommand]
+        addTargetWithHandler:^MPRemoteCommandHandlerStatus(
+            MPRemoteCommandEvent * _Nonnull event) {
+          aud_drct_play_pause();
+          return MPRemoteCommandHandlerStatusSuccess;
+        }];
+    // skip forward
```

Comment:
Use a consistent capitalization for the comments and use (or remove) empty lines between the blocks here.